xc/device.ml

```diff
@@ -608,11 +610,32 @@ let qemu_media_change ~xs device _type params =
 	if is_upstream_qemu device.frontend.domid
 	then begin
 		let cd = "ide1-cd1" in
-		let qmp_cmd =
-			if params = ""
-			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
-			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
-		qmp_write device.frontend.domid qmp_cmd
+		match params with
```

Comment:
> And If we send a command to QEMU, QEMU will response with a Success or Error, so it makes sense to read the response after sending command.
> ```
>  -			if params = ""
>  -			then (Qmp.Command(None, Qmp.Eject (cd, Some true)))
>  -			else (Qmp.Command(None, Qmp.Change (cd, params, None))) in
>  -		qmp_write device.frontend.domid qmp_cmd
> ```
> The original code just sends qmp_cmd without reading response.

If we always need to read after a write, can't we make qmp_write() return the result of a qmp_read() automatically? This will make the code safer, because no one will ever forget to read the result of a write.